employees/views.py

```diff
@@ -9,10 +16,99 @@ class ReportViewSet(viewsets.ModelViewSet):
     serializer_class = ReportSerializer
     permission_classes = (
         permissions.IsAuthenticated,
-        )
+    )
 
     def get_queryset(self):
         return Report.objects.filter(author=self.request.user).order_by('-date')
 
     def perform_create(self, serializer):
         serializer.save(author=self.request.user)
+
+
+def query_as_dict(query_set):
+    dictionary = {}
+    for record in query_set:
+        key = record.date
+        dictionary.setdefault(key, [])
+        dictionary[key].append(record)
+    return dictionary
+
+
+class ReportList(APIView):
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    template_name = 'employees/report_list.html'
+    permission_classes = (
+        permissions.IsAuthenticated,
+    )
+
+    def get_queryset(self):
+        return Report.objects.filter(author=self.request.user).order_by('-date', 'project__name')
+
+    def get(self, request):
+        reports_serializer = ReportSerializer(context={'request': request})
+        reports_dict = query_as_dict(self.get_queryset())
+        return Response({
+            'serializer': reports_serializer,
+            'reports_dict': reports_dict,
+            'UI_text': ReportListStrings,
+        })
+
+    def post(self, request):
+        reports_serializer = ReportSerializer(data=request.data, context={'request': request})
+        reports_dict = query_as_dict(self.get_queryset())
+        if not reports_serializer.is_valid():
+            return Response({
+                'serializer': reports_serializer,
+                'reports_dict': reports_dict,
+                'errors': reports_serializer.errors,
+                'UI_text': ReportListStrings,
+            })
+        reports_serializer.save(author=self.request.user)
+        reports_dict = query_as_dict(self.get_queryset())
+        reports_serializer = ReportSerializer(context={'request': request})
+        return Response({
+            'serializer': reports_serializer,
+            'reports_dict': reports_dict,
+            'UI_text': ReportListStrings,
+        }, status=201)
+        # QUESTION: Should I use some sort of database verification prior to returning Response with status 201?
```

Comment:
I think that we will need to think about it how to do database verification. If any error occurs will `save()` method raise exception? Please investigate it.

Reply:
![image](https://user-images.githubusercontent.com/17569113/52644717-a0b03d00-2edf-11e9-9ca7-a8322dea7992.png)

So after my thorough investigation, it turns out that Django in fact throws an AssertionError if `serializer.save()` is called without `serializer.is_valid()` preceding it. Thus it forces you to perform validation check on serializer before saving it's data in database.
So I sabotaged the serializer to have it pass incomplete data to model. In such case an IntegrityError is thrown if model receives bad data.